### ISSUE TITLE
How to turn off minification?

### ISSUE THREAD

**Issue by ccorcos** [webpack-contrib/html-loader#65]
```
    htmlLoader: {
      minimize: false,
    },
```

doesnt seem to work

**Comment by ccorcos** [comment]
These defaults are terrible:

https://github.com/webpack/html-loader/blob/master/index.js#L71-L88


```
<!DOCTYPE html> <html lang=en> <head> <meta charset=UTF-8> <title>Skeleton App</title> <meta name=viewport content="width=device-width,initial-scale=1"> </head> <body> <div id=root></div> <img src=<%=ROOT_URL%>/7d3f8da0eae1b0de9b3e8ef7ac4615b4.png /> <script type="text/javascript" src="<%=ROOT_URL%>/index-542e410e44a0f3d636e0.js"></script></body> </html>
```

**Comment by ccorcos** [comment]
I think it would be great if they were configurable but I cant figure out how to do that. Perhaps its a porblem with using it inside HtmlWebpackPlugin

**Comment by toverux** [comment]
Use it like that :

```js
{ test: /\.html$/, loader: 'html?minimize=false', exclude: [root('src', 'index.html')] },
```

https://github.com/webpack/html-loader/blob/master/index.js#L68

And yes, I agree, these are very destructive defaults :/

**Comment by ccorcos** [comment]
hmm. the problem is I'm using it with the Webpack HTML Plugin...

**Comment by toverux** [comment]
Me too :) And the HtmlWebpackPlugin does not seem to cause any problem here, it's just html-loader's minimification, at least in my configuration (note that I also have excluded index.html in html-loader configuration):

```js
const BASE_CONFIG = {
    plugins: [
        new HtmlWebpackPlugin({ template: 'src/index.html' })
    ],

    module: {
        loaders: [
            { test: /\.html$/, loader: 'html?minimize=false', exclude: [root('src', 'index.html')] }
        ]
    }
};
```

**Comment by imvetri** [comment]
You can try the following? Please dont mind. i didnt try the code, I could guess this from the code
[https://github.com/webpack/html-loader/blob/master/index.js#L68](url) difference. 

`htmlLoader:{"removeComments : false",`
` "removeCommentsFromCDATA : false",`
` "removeCDATASectionsFromCDATA : false", `
`"collapseWhitespace : false", `
`"conservativeCollapse : false",`
` "removeAttributeQuotes : false", `
`"useShortDoctype : false",`
` "keepClosingSlash : false",`
` "minifyJS : false", `
`"minifyCSS : false", `
`"removeScriptTypeAttributes : false",`
` "removeStyleTypeAttributes : false"}`

**Comment by imvetri** [comment]
@ccorcos Can i request something? Could you share information on How you specifically opened the code difference at [https://github.com/webpack/html-loader/issues/url](url) ?

Above comment is my first comment on a open source  :dancer: yay. and i am curious how you were able to point at specific piece of code.

**Comment by ccorcos** [comment]
So I'm using the html webpack plugin like this:

```js
      new HtmlWebpackPlugin({
        template: `!html!${config.html}`,
        inject: 'body',
      }),
```

It appears that adding any inline query params causes errors, and when I use something like...

```js
    htmlLoader: {
      removeComments: false,
      removeCommentsFromCDATA: false,
      removeCDATASectionsFromCDATA: false,
      collapseWhitespace: false,
      conservativeCollapse: false,
      removeAttributeQuotes: false,
      useShortDoctype: false,
      keepClosingSlash: false,
      minifyJS: false,
      minifyCSS: false,
      removeScriptTypeAttributes: false,
      removeStyleTypeAttributes: false
    },
```

... it doesnt work either.

@imvetri When you are looking at code on github, click a line number and it it will get highlighted and change the url. You can also shift-click to another location to highlight a range.

**Comment by jackyon** [comment]
not working for me either, the html code still get compressed..
is there any solution for this issue?

**Comment by imvetri** [comment]
Can someone help me understand with this condition
https://github.com/webpack/html-loader/blob/master/index.js#L68

I suspect this line and it is taking program control inside the IF block all ways.

 Hence minify=true is not working?

**Comment by abelokon** [comment]
1+ same problem using HtmlWebpackPlugin - any updates?

**Comment by ccorcos** [comment]
I didnt figure anything out -- at then end of the day, it does work...

**Comment by d3viant0ne** [comment]
@abelokon There are a few "problems" mentioned here. The original was the defaults for html loader in regards to minification which has been changed. The second was effectively minifing twice. 

Could you be a bit more specific to which issue you are referencing?

**Comment by abelokon** [comment]
@d3viant0ne As already mentioned I'm also using [HtmlWebpackPlugin](https://github.com/ampedandwired/html-webpack-plugin) and trying figure out how to turn off the minification of my html files. Since the first snippet posted here didn't work for me either, I agreed.

**Comment by eladchen** [comment]
FYI: Looking at this [commit](https://github.com/webpack/html-loader/commit/30db815dfba55dcd647ee2bbd6706550ddef5e4f) @ [index.js:19 ](https://github.com/webpack/html-loader/blob/master/index.js#L19)

It seems that there is an issue with how "configKey" is looked up. The config key is looked up using
hasOwnProperty( ... ) on 'context.options' - This will never yield true, as the key does exists, but on 'context.options.prototype'.

Please correct me if I'm wrong.

**Comment by ghjagus** [comment]
Is there a way that minimize the html string but keep case sensitive?Because i use PascalCase-style to name my components.

**Comment by hemanth** [comment]
Thanks for reporting, this is fixed in #116.

**Comment by amirhosein993** [comment]
> I use it this way, and that works :
> 
> ```js
> { test: /\.html$/, loader: 'html?minimize=false', exclude: [root('src', 'index.html')] },
> ```
> 
> https://github.com/webpack/html-loader/blob/master/index.js#L68
> 
> And yes, I agree, these are very destructive defaults :/

where to write this? 
thanks
